framework/packages/abstract-client/src/lib.rs

```diff
@@ -32,14 +32,83 @@ pub use source::AccountSource;
 mod interchain {
     pub(crate) mod remote_account;
     mod remote_application;
+    use std::collections::HashMap;
+
+    use cw_orch_interchain::{IbcQueryHandler, InterchainEnv, InterchainError};
     pub use remote_account::RemoteAccount;
     pub use remote_application::RemoteApplication;
 
+    use crate::{client::AbstractClientResult, AbstractClient, Environment};
+
     // TODO: Why are we not returning ibc tx analysis after await
     /// IbcTxAnalysis after waiting for interchain action
     pub struct IbcTxAnalysisV2<Chain: cw_orch::environment::CwEnv>(
         pub cw_orch_interchain::types::IbcTxAnalysis<Chain>,
     );
+
+    /// Client to interact with Abstract and setup interchain capabilities
+    pub struct AbstractInterchainClient<Chain: IbcQueryHandler> {
+        /// All clients registered within this Interchain Client
+        pub clients: HashMap<String, AbstractClient<Chain>>,
+    }
+
+    impl<Chain: IbcQueryHandler> AbstractInterchainClient<Chain> {
+        /// Deploys and connects Abstract instances across all chains specified
+        pub fn build<Interchain: InterchainEnv<Chain>>(
+            interchain: &Interchain,
+        ) -> AbstractClientResult<Self> {
```

Comment:
Why is it called `build` if we never get to the builder?

Reply:
What if we duplicate AbstractClientBuilder methods but with "chain_id" passed, so you can edit each abstract individually?